### ISSUE TITLE
WebSocket is not defined (Node + Webpack)

### ISSUE THREAD

**Issue by gempain** [stomp-js/stompjs#119]
Hello,

I am getting the following error in a Node application:
```
STOMP: connected to server ...
STOMP: send PING every 10000ms
STOMP: check PONG every 10000ms
STOMP: Received data
STOMP: <<< PONG
STOMP: Received data
STOMP: <<< PONG

/myproject/node_modules/@stomp/stompjs/bundles/webpack:/StompJs/src/stomp-handler.ts:251
        if (this._webSocket.readyState === WebSocket.OPEN) {
                                           ^
ReferenceError: WebSocket is not defined
    at Timeout._onTimeout (/myproject/node_modules/@stomp/stompjs/bundles/webpack:/StompJs/src/stomp-handler.ts:251:44)
    at listOnTimeout (internal/timers.js:535:17)
    at processTimers (internal/timers.js:479:7)
```

As you can see, the WebSocket is opening just fine, and pings are coming in normally. The error is thrown after a couple of pings have arrived, I'd say 10 seconds after the socket has opened.

`WebSocket` should be defined as I have the following safeguards:
1. At the beginning of the app:
```
import * as WebSocket from 'ws';
```
2. In my Webpack configuration file:
```
  plugins: [
    new webpack.ProvidePlugin({
      WebSocket: 'ws'
    }),
  ],
```

I currently resolve this by following the workaround mentioned in a (most likely) [related issue](https://github.com/stomp-js/stompjs/issues/28):
```
Object.assign(global, { WebSocket: require('ws') });
```

All the best, and keep up the great work !

**Comment by kum-deepak** [comment]
Thanks for the workaround. While this itself may not be a bug, the code should not refer to WebSocket as WebSocket is technically not needed to use this library.

**Comment by gempain** [comment]
@kum-deepak very welcome ! I was surprised to see that the object was referenced globally. Why not use a reference to `this._wesocket` which seems to be referenced from the code in many places ?

**Comment by kum-deepak** [comment]
Fixed and released in 5.4.2.

**Comment by MuhammadShaban** [comment]
still exists

**Comment by kum-deepak** [comment]
Please share details about your setup and copy of console log.

**Comment by MuhammadShaban** [comment]
(node:14436) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:14436) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.   
(node:14436) UnhandledPromiseRejectionWarning: ReferenceError: WebSocket is not defined
    at Client../src/client.ts.Client._createWebSocket (C:\Users\m.shaaban\Cequens\Projects\sms_http_bulk_interface\node_modules\@stomp\stompjs\bundles\stomp.umd.js:419:13)
    at Client.<anonymous> (C:\Users\m.shaaban\Cequens\Projects\sms_http_bulk_interface\node_modules\@stomp\stompjs\bundles\stomp.umd.js:359:48)
    at step (C:\Users\m.shaaban\Cequens\Projects\sms_http_bulk_interface\node_modules\@stomp\stompjs\bundles\stomp.umd.js:165:23)
    at Object.next (C:\Users\m.shaaban\Cequens\Projects\sms_http_bulk_interface\node_modules\@stomp\stompjs\bundles\stomp.umd.js:146:53)
    at fulfilled (C:\Users\m.shaaban\Cequens\Projects\sms_http_bulk_interface\node_modules\@stomp\stompjs\bundles\stomp.umd.js:137:58)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
(node:14436) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)

**Comment by kum-deepak** [comment]
Please check https://stomp-js.github.io/faqs/2019/05/20/faqs.html#p-can-this-library-be-used-with-nodejs-p

If that does not help, please share your Stomp Config.

**Comment by MuhammadShaban** [comment]
i'm already solve this problem by using this workaround
Object.assign(global, {
    WebSocket: websocket.w3cwebsocket,
    // Not needed in node 11
    TextEncoder: TextEncodingPolyfill.TextEncoder,
    TextDecoder: TextEncodingPolyfill.TextDecoder
});

but when i'm seeing your commit 'Fixed and released in 5.4.2.'
i think that i can use it without this workaround
